### PR TITLE
[TT-9169/TT-9341] update docs

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description

Add config doc to let users know `disable_key_actions_by_username` should the same in dashboard

## Related Issue

https://tyktech.atlassian.net/browse/TT-9341

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5264/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5453218615)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5264/merge
Commit: c8479ebf9c77ae071b54ca1cd0e13dfdab618802 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5453392861)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5264/merge
Commit: 02fc3b400a1bb09cd082b83d88b2208aed81013e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5461580616)

File: config/config.go [hunk 1]
```diff
@@ -637,6 +637,7 @@ type Config struct {
 	HashKeys bool `json:"hash_keys"`
 
 	// DisableKeyActionsByUsername disables key search by username.
+	// Please ensure that dashboard has the same value for seamless functionality.
```
Comment (andyo-tyk) [inline config/config.go:640]:
```suggestion
	// When this is set to `true` you are able to search for keys only by keyID or key hash (if `hash_keys` is also set to `true`)
	// If you are using the Tyk Dashboard, you must configure this setting with the same value in both Gateway and Dashboard
```

Comment (andyo-tyk) [reply config/config.go:640]:
```suggestion
	// When this is set to `true` you are able to search for keys only by keyID or key hash (if `hash_keys` is also set to `true`)
	// Note that if `hash_keys` is also set to `true`, then the keyID will not be provided for APIs secured using basic auth so the only search option would be using key hash
	// If you are using the Tyk Dashboard, you must configure this setting with the same value in both Gateway and Dashboard
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Updated to clarify behaviour with basic auth and hashed keys

**Review by dcs3spp — APPROVED** [review]
lgtm

File: config/config.go [hunk 2]
```diff
@@ -637,6 +637,9 @@ type Config struct {
 	HashKeys bool `json:"hash_keys"`
 
 	// DisableKeyActionsByUsername disables key search by username.
+	// When this is set to `true` you are able to search for keys only by keyID or key hash (if `hash_keys` is also set to `true`)
+	// Note that if `hash_keys` is also set to `true`, then the keyID will not be provided for APIs secured using basic auth so the only search option would be using key hash
```
Comment (andyo-tyk) [inline config/config.go:641]:
```suggestion
	// Note that if `hash_keys` is also set to `true` then the keyID will not be provided for APIs secured using basic auth. In this scenario the only search option would be to use key hash
```

**Review by andyo-tyk — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5264/merge
Commit: 088a11e884421e1b0122f3329597e730a172f276 
Triggered by: pull_request (@andyo-tyk)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5486887772)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5264/merge
Commit: 01cfc0a6692da750dce54792b42387b742d5f06e 
Triggered by: pull_request (@lghiur)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5487198515)

**Pull request merged**